llvm/lib/Target/AMDGPU/SIInstrInfo.td

```diff
@@ -2904,6 +2905,15 @@ def getVCMPXOpFromVCMP : InstrMapping {
   let ValueCols = [["1"]];
 }
 
+def FP8DstByteSelTable : GenericTable {
+  let FilterClass = "VOP3_Pseudo";
```

Comment:
I think, its better to filter it on encoded variants of the instruction. For example, V_CVT_SR_FP8_F32 is also defined for _gfx12_ which also uses _VOP3_Pseudo_. For gfx12, V_CVT_SR_FP8_F32 
 uses _VOP3_CVT_SR_F8_ByteSel_Profile_ but _HasFP8DstByteSel_ is _false_ there that makes _HasFP8DstByteSel_ inconsistent. I am not sure whether this hazard needs to be resolved for gfx12 or not (I assume NO)

Reply:
In general the w/a is already guarded by the ST.hasDstSelForwardingHazard() check, which is limited to gfx940.

Reply:
We need to resolve hazard for mi300 and mi350 (at least). For gfx12 I think there is no issue, but `HasFP8DstByteSel` should be consistent -- it's behavior shouldn't be reliant upon the current users of it.

I've resolved the inconsistency by modifying VOP3_CVT_SR_F8_ByteSel_Profile

Reply:
> I think, its better to filter it on encoded variants of the instruction

I'm not sure there is any reason to consider the encoded variants / I think VOP3_Pseudo filter is fine. Based on my understanding, the VOP3_Pseudos will be used during CodeGen / HazardRecognizer, and the encoded variants / opcodes aren't introduced until MCLayer.

Reply:
@arsenm 

Do you think filtering in VOP3_Pseudo is sufficient, or should we also also include the encoded variants in the generated table?

Reply:
Ping

#I think, its better to filter it on encoded variants of the instruction

@pravinjagtap I'm not sure I understand why we need to include the encoding opcodes -- why do we need to also include these in our generated table ?

Reply:
VOP3_Pseudo is sufficient for now